bbn-test-5/finality-provider/upgrades/v2rc0/README.md

```diff
@@ -0,0 +1,145 @@
+# `v2.0.0-rc.0` Software Upgrade
+
+This document summarizes the procedure to upgrade a Babylon Finality Provider
+to version v2.0.0-rc.0.
+
+## Table of Contents
+
+1. [High-level Overview](#1-high-level-overview)
+2. [Technical Deep-dive](#2-technical-deep-dive)
+3. [Applying the upgrade](#3-applying-the-upgrade)
+   1. [Preparation](#31-preparation)
+   2. [Execution](#32-execution)
+   3. [Verification](#33-verification)
+
+## 1. High-level Overview
+
+- **Upgrade version**: `v2.0.0-rc.0`
+- **Upgrade Babylon height**: `X` (TODO: specify the height)
+- **Upgrade overview**: This upgrade introduces a major refactoring that
+  transforms the Finality Provider into a reusable SDK, to enable the
+  development of Finality Provider software for BSNs.
+- **Finality Providers who should participate**: Only Babylon Finality Providers.
+
+## 2. Technical Deep-dive
+
+The Babylon Genesis Testnet `v3.0.0-rc.0` upgrade (TODO: link to corresponding
+upgrade doc) is coupled with a coordinated breaking upgrade of the Finality
+Provider software from `v1.x.x` (`v1.0.0`, `v1.1.0-rc.0`, `v1.1.0-rc.1`) to
+`v2.0.0-rc.0`.
+
+Finality Provider `v2.0.0-rc.0` introduces a major refactoring of the codebase
+into a reusable SDK. This aims to enable the development of Finality
+Provider software for Bitcoin Supercharged Networks (BSNs). The following BSN
+stacks are initially supported:
+- Ethereum L2 Rollup BSNs, integrating via the deployment of a CosmWasm Smart
+  Contract on the Babylon Genesis Testnet. The Rollup BSN Finality Provider
+  will be responsible for the submission of finality signatures and public
+  randomness to the Contract.
+- Cosmos SDK BSNs, integrating via the deployment of a CosmWasm Smart Contract
+  on their respective chains and the establishment of IBC Channels with Babylon.
+  The Cosmos BSN Finality Provider will be responsible for the submission of
+  finality signature and public randomness to the Contract.
```

Comment:
It's unclear whether the above is directly relevant to Babylon Genesis finality providers, especially since they won't be using the BSN software. It might be more effective to include a concise bullet list (similar to what I suggested earlier for the previous sectoin) that outlines our key focus areas. This should also emphasize our efforts around compatibility and, importantly, the signing context.

Given our concerns around the signing context, it would make sense to concentrate on technically expanding its functionality. We could also highlight potential pitfalls that implementers might face, so they’re well-prepared in advance.